pkg/runtime/serializer/recognizer/recognizer.go

```diff
@@ -27,18 +27,38 @@ import (
 
 type RecognizingDecoder interface {
 	runtime.Decoder
+	// AccurateRecognizer should be true if this decoder can unambigously determine
+	// whether the content is known to it. Allows RecognizingDecoders to adjust certainty.
```

Comment:
Please clarify if this means it allows false positives, false negatives, or both.